### PR TITLE
feat: stream configuration redesign with named streams and multi-protocol support

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

- **Required stream names** for identification and Home Assistant MQTT integration
- **Five stream types supported**: RTSP, HTTP, HLS, RTMP, UDP/RTP
- **Per-stream transport protocol** (TCP/UDP) instead of global setting
- **Automatic migration** from legacy `urls[]` format to new `streams[]` format
- **Stream health API** now includes name and type in responses

### Backend (Go)
- Add `StreamConfig` type with name, URL, type, and transport fields
- Add validation for stream names (required, max 64 chars, unique) and URL schemes
- Add `MigrateRTSPConfig()` for backward-compatible migration with type inference
- Update stream health API responses to include name and type

### Frontend (Svelte/TypeScript)
- Add TypeScript interfaces matching Go structs
- Update StreamCard with name field and all 5 stream types
- Update StreamManager with name validation and auto-detection
- Add i18n translations for new UI elements

### Configuration
- Update default config template with new streams format and examples

## Test plan

- [x] All stream validation tests pass (17 tests)
- [x] All migration tests pass including mixed URL types
- [x] Frontend TypeScript compiles with no errors
- [x] Frontend lint passes (0 errors)
- [x] Code review fixes applied (stale settings, icon colors, parallel tests)

## Summary by CodeRabbit

* **New Features**
  * Per-stream configuration: name, protocol auto-detection (RTSP/HTTP/HLS/RTMP/UDP), per-stream transport, stable icon colors, masked credentials, and per-stream health/status.

* **Bug Fixes / Validation**
  * Stronger validation: name/URL required, duplicate-name/URL checks, type-aware URL validation, and disabled save when invalid.

* **API Changes**
  * Stream health responses now include stream name and type for clearer status reporting.

* **Chores**
  * Automatic migration from legacy RTSP urls/transport to streams; config and docs updated.

* **Tests**
  * Extensive migration and validation test coverage added.

* **Localization**
  * Added/updated stream UI copy and error messages across locales.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: internal/conf/stream_migration_test.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+				},
+			},
+			expectMigrated: true,
+			expectedCount:  1,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/conf/stream_migration_test.go:90]: This project uses Go 1.25.3 (see go.mod). Since Go 1.22 (Feb 2024), each loop iteration gets its own variable copy, making the `tt := tt` capture unnecessary. See: https://go.dev/blog/loopvar-preview

The referenced Go wiki page predates this change.

File: internal/conf/stream_validation_test.go [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+				Type: StreamTypeHTTP,
+			},
+			wantErr: true,
+			errMsg:  "HTTP type requires http://",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/conf/stream_validation_test.go:169]: Same as above - Go 1.22+ fixed loop variable scoping. This project uses Go 1.25.3, so no capture needed.

File: frontend/src/lib/desktop/components/forms/StreamCard.svelte [hunk 3]
```diff
@@ -163,8 +176,14 @@
...
+      onUpdate({
+        name: editName.trim(),
+        url: editUrl.trim(),
+        type: editStreamType,
+        // Use selected transport for RTSP/RTMP, default 'tcp' for others (ignored by backend)
+        transport: showTransportInEdit ? editTransport : 'tcp',
+      });
       isEditing = false;
     }
   }
```
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/src/lib/desktop/components/forms/StreamCard.svelte:179]: Addressed in commit 46c0d70 - onUpdate now returns boolean and saveEdit only closes edit mode on success.

File: frontend/src/lib/desktop/components/forms/StreamManager.svelte [hunk 4]
```diff
@@ -239,84 +242,161 @@
...
+      // Update health map if URL changed
+      if (oldStream && oldStream.url !== updatedStream.url) {
+        streamHealth.delete(oldStream.url);
       }
-      streamTransports.set(url, streamTransport);
+
+      updatedStreams.splice(index, 1, updatedStream);
+      onUpdateStreams(updatedStreams);
     }
   }
```
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/src/lib/desktop/components/forms/StreamManager.svelte]: Addressed in commit 46c0d70 - updateStream now returns boolean and shows toast notifications on validation failures.

File: frontend/src/lib/desktop/components/forms/StreamCard.svelte [hunk 5]
```diff
@@ -53,18 +50,19 @@
...
     { value: 'rtsp', label: 'RTSP' },
-    // Future options - disabled for now
-    // { value: 'http', label: 'HTTP', disabled: true },
-    // { value: 'hls', label: 'HLS', disabled: true },
-    // { value: 'rtmp', label: 'RTMP', disabled: true },
+    { value: 'http', label: 'HTTP' },
+    { value: 'hls', label: 'HLS' },
+    { value: 'rtmp', label: 'RTMP' },
+    { value: 'udp', label: 'UDP/RTP' },
   ];
```
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/src/lib/desktop/components/forms/StreamCard.svelte:67]: Good catch. The labels are currently hardcoded as they match standard protocol abbreviations (RTSP, HTTP, etc.) that are typically not localized. However, I can add i18n keys if the project prefers localized protocol labels. Would you like me to add this in a follow-up?

File: internal/conf/validate.go [hunk 6]
```diff
@@ -74,6 +88,97 @@ type ValidationResult struct {
...
+			return fmt.Errorf("stream '%s': RTMP type requires rtmp:// or rtmps:// URL", s.Name)
+		}
+	case StreamTypeUDP:
+		if !strings.HasPrefix(urlLower, "udp://") && !strings.HasPrefix(urlLower, "rtp://") {
+			return fmt.Errorf("stream '%s': UDP type requires udp:// or rtp:// URL", s.Name)
+		}
+	}
+
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/conf/validate.go:151]: Valid suggestion. The current implementation uses strings.HasPrefix which works correctly for well-formed URLs. Using net/url.Parse would add robustness for edge cases. I'll consider this for a follow-up improvement, though the current approach passes all validation tests and handles the expected URL formats correctly.

File: frontend/static/messages/en.json [hunk 7]
```diff
@@ -1951,17 +1954,19 @@
...
-          "description": "Add RTSP streams to capture audio from network cameras or other RTSP sources.",
+          "description": "Add audio streams from IP cameras, Icecast servers, or other network sources.",
           "hintsTitle": "Getting Started",
           "hints": {
-            "rtsp": "Use RTSP URLs from IP cameras (e.g., rtsp://192.168.1.100:554/stream)",
+            "rtsp": "Supports RTSP, HTTP, HLS, RTMP, and UDP streams",
             "credentials": "Include credentials in URL if required (rtsp://user:pass@host/stream)",
-            "protocol": "TCP is more reliable, UDP has lower latency"
+            "protocol": "TCP is more reliable for RTSP, UDP has lower latency"
           }
```
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/static/messages/en.json]: Fixed. Removed credential-like URL examples from emptyState.hints.credentials in all 8 translation files.

File: internal/conf/config.go [hunk 8]
```diff
@@ -1523,6 +1558,102 @@ func (s *Settings) MigrateOAuthConfig() bool {
...
+	streamIndex := 0
+
+	// Migrate each URL to StreamConfig
+	for _, rawURL := range rtsp.URLs {
+		// Trim whitespace and skip empty URLs
+		url := strings.TrimSpace(rawURL)
+		if url == "" {
+			continue
+		}
+		streamIndex++
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/conf/config.go:1629]: Fixed. Added duplicate URL detection to migration - uses a map to track seen URLs and skips duplicates to ensure valid configuration.

File: internal/api/v2/streams_health.go [hunk 9]
```diff
@@ -135,6 +140,31 @@ func (c *Controller) initStreamHealthRoutes() {
...
+	}
+
+	for _, stream := range settings.Realtime.RTSP.Streams {
+		if stream.URL == rawURL {
+			return streamInfo{Name: stream.Name, Type: stream.Type}
+		}
+	}
+
+	return streamInfo{}
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/api/v2/streams_health.go:166]: This is an edge case. The frontend already handles ambiguous matching via name-based lookup (implemented in commit 46c0d70). 

The scenario where this fails:
1. User changes credentials in config
2. FFmpeg process is still running with old URL
3. Backend lookup fails

However, credential changes require service restart anyway for FFmpeg to reconnect with new credentials. The mismatch only exists during runtime-config divergence which is transitional. Will consider URL normalization as a follow-up hardening if this becomes a reported issue.
